server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -917,9 +918,16 @@ public int hashCode() {
         }
     }
 
+    /**
+     * Task that runs on the master node. Handles responding to the request listener with the result of the update request.
+     * Task is created when the master node receives a data node request to mark a shard as STARTED.
```

Comment:
Maybe clarify which `STARTED` we mean here (needs an `import` too):

```suggestion
     * Task is created when the master node receives a data node request to mark a shard as {@link ShardRoutingState#STARTED}.
```